### DISCUSSION TITLE
Why do You use records for a Customer & Account models / aggregates?

### DISCUSSION THREAD

**Discussion by patryk-roguszewski** [mizrael/SuperSafeBank#11] [General]
Why do You use records for Customer & Account models/aggregates? 

Records are immutable by the definition, but I see You are using it for entities/models/aggregates where their state can be changed. Is There a special reason for that kind of implementation?

**Comment by mizrael** [comment]
hi @don-flamingo , thanks for reaching out! No particular reason actually. I find them very useful when implementing Value Objects and I kept that as a habit. I like the idea of the equality comparers being semi-magically auto-implemented (although I rarely make use of it)
